### ISSUE TITLE
Uses deprecated string-similarity@4.0.4

### ISSUE THREAD

**Issue by jmrnilsson** [elastic/apm-agent-nodejs#3588]
**Describe the bug**

Installing this package yields a message:

```txt
deprecated [string-similarity@4.0.4](mailto:string-similarity@4.0.4): Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
```

**To Reproduce**

Install the package.
See above mentioned message.
(Or just run npm audit).

**Expected behavior**

Should not be relying on deprecated packages.

**Additional context**
string similarity supposedly uses a Dice coefficient rathen than Levenstein. Found this library that does a similar thing.
https://www.npmjs.com/package/dice-coefficient

The message suggest reaching out to nom support.

**Comment by jmrnilsson** [comment]
Similar to: https://github.com/elastic/apm-agent-nodejs/issues/2644

In v.3.47.0.

**Comment by jmrnilsson** [comment]
Checking if fixed in 3.49.0.

**Comment by david-luna** [comment]
Hi @jmrnilsson 

unfortunately we cannot bump the version due to node support versions. For more info check https://github.com/elastic/ecs-logging-nodejs/pull/145#issuecomment-1544472161

**Comment by jmrnilsson** [comment]
It's fixed in 3.49.1. Closing.
